src/java.xml/share/classes/javax/xml/xpath/XPathFactory.java

```diff
@@ -369,4 +369,58 @@ public abstract boolean getFeature(String name)
     * @return New instance of an <code>XPath</code>.
     */
     public abstract XPath newXPath();
+
+    /**
+     * Sets a property for this {@code XPathFactory} and {@code XPath}
+     * created by this factory.
+     *
+     * @implSpec
+     * The default implementation throws
+     * {@link java.lang.UnsupportedOperationException}.
+     *
+     * @param name the property name
+     * @param value the value for the property
+     *
+     * @throws IllegalArgumentException if the property name is not recognized,
+     * or the value can not be assigned
+     * @throws UnsupportedOperationException if the implementation does not
+     * support the method
+     * @throws NullPointerException if the {@code name} is {@code null}.
+     *
+     * @since 18
+     */
+    public void setProperty(String name, String value)
+        throws IllegalArgumentException {
+
+        if (name == null) {
+            throw new NullPointerException("the name parameter is null");
+        }
+        throw new UnsupportedOperationException("not implemented");
+    }
+
+    /**
+     * Returns the value of the specified property.
+     *
+     * @implSpec
+     * The default implementation throws
+     * {@link java.lang.UnsupportedOperationException}.
+     *
+     * @param name the property name
+     * @return the value of the property.
+     *
+     * @throws IllegalArgumentException if the property name is not recognized
+     * @throws UnsupportedOperationException if the implementation does not
+     * support the method
+     * @throws NullPointerException if the {@code name} is {@code null}.
+     *
+     * @since 18
+     */
+    public String getProperty(String name)
+        throws IllegalArgumentException {
+
+        if (name == null) {
+            throw new NullPointerException("the name parameter is null");
+        }
+        throw new UnsupportedOperationException("not implemented");
+    }
```

Comment:
Changing it to throw IAE looks much better.  You can drop "throws IllegalArgumentException" from the method signature as it's a runtime exception.

Reply:
IAE dropped.